packages/animations/browser/src/util.ts

```
@@ -159,14 +160,18 @@ function writeStyleAttribute(element: any) {
   element.setAttribute('style', styleAttrValue);
 }
 
-export function setStyles(element: any, styles: ɵStyleData, formerStyles?: {[key: string]: any}) {
+export function setStyles(
+    element: any, styles: ɵStyleData, normalizer: AnimationStyleNormalizer,
+    formerStyles?: {[key: string]: any}) {
   if (element['style']) {
     Object.keys(styles).forEach(prop => {
       const camelProp = dashCaseToCamelCase(prop);
       if (formerStyles && !formerStyles.hasOwnProperty(prop)) {
         formerStyles[prop] = element.style[camelProp];
       }
-      element.style[camelProp] = styles[prop];
+      const normalizedProp = normalizer.normalizePropertyName(prop, []);
+      element.style[camelProp] =
+          normalizer.normalizeStyleValue(prop, normalizedProp, styles[prop], []);
```

@JoostK sorry for bothering you :cry: 

Could you advise on what to do with the errors here? 
(I am currently ignoring them since I just pass empty one-time arrays)

Should I gather the errors and throw as in: https://github.com/angular/angular/blob/c9b47f21822343286d056fd70aec882acc0ae788/packages/animations/browser/src/render/shared.ts#L82 ?

---

That is a very good question. There is currently one error reported for the case where a string without unit is provided for a dimensional property. I think reporting that is reasonable, however that would introduce a breaking change and limits this change from landing in patch.

Therefore, I think it's reasonable to ignore the errors for now such that the behavior is not affected. We can later revisit this choice in a major version if needed. It may depend a bit on how breaky it turns out in Google.

---

We may want to also have a test for the behavior we choose here; i.e. that a string without number bound to a dimension property applies the string verbatim without an error.

---

About the breaking change thing, I understand and yeah I would really like it to just get this fixed now and add the error handling later :slightly_smiling_face: 

would you mind if I add a comment there? 
(something like: `// TODO: (try to) add error handling before new major release (introduces a breaking change)`)

---

About the test, I actually tried to implement it so that but I actually think that the animations don't past a certain point if they don't conform to the standards

to be more clear, I tried to change my test and modified 300 to "300", when running it I got:
```
Error: Unable to animate due to the following errors:
         - Please provide a CSS unit value for width:300
```         
         
So I looked into it and it's the error thrown in `normalizeKeyframes` (the same I linked in my comment), that is called as part of the `TransitionAnimationEngine`'s `flush` method

So if I understood correctly, the `flush` happens still as part of the animation before the `destroy` to which I am mainly hooking into, so if there were actual errors in the values they would have been caught by `flush`, so I think that my ignoring those errors is actually correct as there may not be errors at this point, because if there were errors they would have already been caught and an error would have been thrown.....

at least that's my understanding of it (if you want I can keep looking into it and try to get more confidence in this)

does this checks out with you? what do you think?

---

I see, thanks for checking. Based on your comment, I realize that the styles do get normalized at some point, but apparently not for all usages. That is an inconsistency in how the styles are interpreted, which is what you are addressing by normalizing the values as they are written.

I was interested to know where the `instruction.toStyles` came from for which the normalization was necessary. These styles originate here:

https://github.com/angular/angular/blob/94e3d3ab70402629fb5882523bc2506d9551dac0/packages/animations/browser/src/dsl/animation_transition_factory.ts#L46-L49

Which calls into the following:

https://github.com/angular/angular/blob/94e3d3ab70402629fb5882523bc2506d9551dac0/packages/animations/browser/src/dsl/animation_transition_factory.ts#L104-L126

I feel like that would actually be the appropriate place to do the normalization as it's still during transaction building where errors can be reported, rather than during write (which creates the unfortunate situation where errors are no longer expected, hence the questions about how to deal with the errors). I think you can get the normalizer in `buildStyles` by adding it to `AnimationTransitionFactory.build`, `AnimationTransitionFactory.buildStyles` and `AnimationStateStyles.buildStyles`. It should then hopefully be possible to normalize the value that is written to `finalStyles`.

---

Oh gosh, I am so sorry you had to look into things so much because of me :cry: 

I see what you mean, I will check it and try to place the normalization where it makes more sense :slightly_smiling_face: 

( again sorry for not having looked at this well enough :sob: )

---

Please, don't be sorry! This is how both of us get to learn something new :)